src/steinertree/steiner_tree.jl

```diff
@@ -0,0 +1,84 @@
+"""
+    filter_non_term_leaves!(g, term_vert)
+
+Remove edges of `g` so that all non-isolated leaves of `g` are in the set `term_vert`
+"""
+function filter_non_term_leaves!(
+    g::AbstractGraph{T},
+    term_vert::Vector{<:Integer}
+    ) where T<:Integer
+
+    is_term = falses(nv(g))
+    is_term[term_vert] .= true
```

Comment:
no need for `.=`,  `=` is sufficient

Reply:
But I'm getting an error when I try that.

```
julia> a = zeros(Bool, 5)
5-element Array{Bool,1}:
 false
 false
 false
 false
 false

julia> b = [2, 4]
2-element Array{Int64,1}:
 2
 4

julia> a[b] = true
ERROR: MethodError: no method matching setindex_shape_check(::Bool, ::Int64)
```

Reply:
Ah my fault, I overlooked that `term_vert` is not a single index.